### PR TITLE
Makes openvex action manual trigger only

### PR DESCRIPTION
**Pull request by psschwei**
`#SecuritySlam`

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This PR is a follow-up to #1465.

Instead of automatically running the openvex action whenever a new tag is created, this PR makes the openvex action require a manual trigger. 

The intention is to better integrate the action into the existing release workflow. To that end, a task was added to the new release template to run this task AFTER the release has been cut (similar to the approach in #1467). The action will then generate the openvex data and upload it to the release. 

The upload step is new in this PR.

#### Which issue(s) this PR fixes:

This commit is part of the December 2023 Security Slam.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

/sig release
/assign @ArangoGutierrez

### CODE REVIEW

File: .github/workflows/openvex.yaml [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
         name: Run vexctl
         with:
-            product: pkg:golang/sigs.k8s.io/kueue@${{ env.RELEASE_VERSION }}
+          product: pkg:golang/sigs.k8s.io/kueue@${{ env.TAG }}
+          file: /tmp/kueve.vex.json
+      - name: Upload SBOM
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh release upload $TAG /tmp/kueve.vex.json
```
Comment (ArangoGutierrez) [inline .github/workflows/openvex.yaml:28]: @alculquicondor @tenzen-y  What do you guys think about this last step?

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md [hunk 2]
```diff
@@ -35,13 +35,14 @@ Please do not remove items from the checklist
...
       `gcr.io/k8s-staging-kueue/kueue:$VERSION`
-- [ ] Submit a PR against [k8s.io](https://github.com/kubernetes/k8s.io), 
+- [ ] Submit a PR against [k8s.io](https://github.com/kubernetes/k8s.io),
       updating `k8s.gcr.io/images/k8s-staging-kueue/images.yaml` to
       [promote the container images](https://github.com/kubernetes/k8s.io/tree/main/k8s.gcr.io#image-promoter)
       to production: <!-- example kubernetes/k8s.io#3612-->
 - [ ] Wait for the PR to be merged and verify that the image `registry.k8s.io/kueue/kueue:$VERSION` is available.
 - [ ] Publish the draft release prepared at the [Github releases page](https://github.com/kubernetes-sigs/kueue/releases).
       Link: <!-- example https://github.com/kubernetes-sigs/kueue/releases/tag/v0.1.0 -->
+- [ ] Run the [openvex action](https://github.com/kubernetes-sigs/kueue/actions/workflows/openvex.yaml) to generate openvex data and add it to the release.
```
Comment (alculquicondor) [inline .github/ISSUE_TEMPLATE/NEW_RELEASE.md:45]:
```suggestion
- [ ] Run the [openvex action](https://github.com/kubernetes-sigs/kueue/actions/workflows/openvex.yaml) to generate openvex data. The action will add the file to the release artifacts.
```
?

File: .github/workflows/openvex.yaml (hunk 1 above)
Comment (alculquicondor) [reply .github/workflows/openvex.yaml:28]: seems fine. Do you have any concerns?

**Review by alculquicondor — COMMENTED** [review]
/approve

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Pull request merged**